### PR TITLE
Optimizer readme

### PR DESCRIPTION
**Pull request by Louis-Dupont**
The requirements were as follows:
- SG Optimizers
- How to use PyTorch native optimizers
- How to use your own optimizers (including in a recipe)

Concerning "How to use your own optimizers": we currently don't support it because the only way to use an optimizer is to pass it as a string, and we don't have a registry for optimizers. Do we want to add this registry ?
Besides this, I don't see what else to show in this notebook, if you have an idea.

### CODE REVIEW

**Review by shaydeci — CHANGES_REQUESTED** [review]
Looks good, I think you are on a good point here - we do need to add registry for optimisers.
Lets add it and update how one can use their own optimiser like you suggested.

**Review by ofrimasad — COMMENTED** [review]
we do have a register_optimizer function now. so you can add this as well

**Review by shaydeci — CHANGES_REQUESTED** [review]
Also another very important point:

We need to add docs for initialise_param_groups and update_param_groups, and demonstrate how it interacts with the optimiser. Thats the only way now we have in SG to support different configurations among different the different network parameters.

File: documentation/source/optimizers.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Optimizers
+
+Optimization is a critical step in the deep learning process as it determines how well the network will learn from the training data.
+SuperGradients supports out-of-the-box pytorch optimizers(
+[SGD](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html#torch.optim.SGD), 
+[Adam](https://pytorch.org/docs/stable/generated/torch.optim.Adam.html#torch.optim.Adam) and 
+[AdamW](https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html#torch.optim.AdamW)
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply documentation/source/optimizers.md]: added

**Comment by Louis-Dupont** [comment]
> Also another very important point:
> 
> We need to add docs for initialise_param_groups and update_param_groups, and demonstrate how it interacts with the optimiser. Thats the only way now we have in SG to support different configurations among different the different network parameters.

I added a section, do you think its enough?

File: documentation/source/optimizers.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+## Customize learning rate for different model blocks 
+You can define the learning rate to use on each section of your model by working with `initialize_param_groups` and `update_param_groups`.
+- `initialize_param_groups` defines the groups, and the learning rate to use for each group. It is called on instantiation.
+- `update_param_groups` updates the learning rate for each group. It is called by LR callbacks (such as `LRCallbackBase`) during the training. 
+
+If your model (i.e. any `torch.nn.Module`) is lacking these methods, the same learning rate will be applied to every block.
+But if you implement them, it will be taken into account by the Trainer just like with any other SuperGradients model.
+
+#### Example
+Let's imagine that you have your own custom model and that you want work with a different learning rate on the backbone.
```
Comment (ofrimasad) [inline documentation/source/optimizers.md:87]: Instead of "let's imagine that you ... ". please write "Assume you have your own..."

**Review by ofrimasad — COMMENTED** [review]
Looks very nice.
I think there is a minor issue in the code

File: documentation/source/optimizers.md (hunk 2 above)
Comment (Louis-Dupont) [reply documentation/source/optimizers.md]: Changed in d66358e09ade583670912a7c5815a0d7cbd4300a

File: documentation/source/optimizers.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            "named_params": self.backbone.named_parameters(),
+            "lr": lr * training_params['multiply_backbone_lr'] # You can use any parameter, just make sure to define it when you set up training_params
+        }
+
+        # OPTIMIZE MAIN ARCHITECTURE LAYERS WITH 10x LR
+        decoder_named_params = list(self.decoder.named_parameters())
+        aux_head_named_parameters = list(self.aux_head.named_parameters())
+        layers_params = {
+            "named_params": decoder_named_params + aux_head_named_parameters,
+            "lr": lr
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply documentation/source/optimizers.md:114]: Fixed in d66358e09ade583670912a7c5815a0d7cbd4300a

**Review by ofrimasad — APPROVED** [review]
LGTM!

**Review by shaydeci — APPROVED** [review]

**Pull request merged**